pkg/activator/handler/handler.go

```diff
@@ -101,11 +103,19 @@ func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 					a.Logger.Warnw("Pod probe failed", zap.Error(err))
 					return false, nil
 				}
+				defer probeResp.Body.Close()
 				httpStatus = probeResp.StatusCode
 				if httpStatus == http.StatusServiceUnavailable {
 					a.Logger.Warnf("Pod probe sent status: %d", httpStatus)
 					return false, nil
 				}
+				if body, err := ioutil.ReadAll(probeResp.Body); err != nil {
+					a.Logger.Warnw("Pod probe returns an invalid response body", zap.Error(err))
```

Comment:
Should this be an `Error`?

Reply:
> I'm a little confused with the networking behavior. We're sending the request to the Kubernetes Service itself, right? We've seen behavior contrary to this when trying to implement active handoff in the autoscaler: Probing the K8s service of the revision from the autoscaler never sent a request to the activator but always to the queue-proxies themselves.
> 
> Are we saying here, that the K8s service is proxied either to activator or queue-proxies?

I think it's because `Host:       r.Host,` points to the public one (e.g. `svc-name.default.example.org`), which will be routed back to activator, while reprogramming hasn't finished.

Reply:
You're right, that case should never happen, and something must be really wrong.

Also, I changed a.Logger.Warnf("Pod probe did not reach the target queue proxy.  Reached: %s", body) to Infof, because this is a normal case.